### PR TITLE
Upgrade and adapt to the new GOBL Extensions feature

### PR DESCRIPTION
**Pull request by cavalle**
* This PR adapts the codebase to use the new IT extensions added in https://github.com/invopop/gobl/pull/184
* Also adapts to other changes in GOBL since the last upgrade
* Adds a test for the item details and totals conversion 
* ✅ <del>🚧 **WIP** 🚧 Don't forget to upgrade the `gobl` module here once the changes in the PR above are released</del>

### CODE REVIEW

File: items_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package fatturapa_test
```
Comment (cavalle) [inline items_test.go:1]: I added these tests to cover the preexisting functionality in `items.go`. I made sure the tests passed in main, before the rest of the changes in this PR.

File: test/data/invoice-hotel.json [hunk 2]
```diff
@@ -117,30 +120,30 @@
...
 			"tax": "11.36",
 			"total_with_tax": "126.00",
 			"payable": "126.00",
 			"advance": "29.00",
 			"due": "97.00"
 		}
-	},
-	"sigs": [
-		"eyJhbGciOiJFUzI1NiIsImtpZCI6ImZmNzU5MTVmLWY2ZjgtNDU4Ny04OTBhLWNjZDhiODZiZjhmNSJ9.eyJ1dWlkIjoiNjc5YTJmMjUtNzQ4My0xMWVjLTk3MjItN2VhMmNiNDM2ZmY2IiwiZGlnIjp7ImFsZyI6InNoYTI1NiIsInZhbCI6ImQzNmM3NGUxOWJmNWYwMDk3NzM4Y2JmNTk3MGZiMjA1NDc5MmU5NDRiMmM3MWMwOWFmMGJjY2I0MmQzZGI1MzYifX0.iivLe0zUAMqlRmN4AJYgUIDhGcyBzYDIBJyTUj0sHBK3axNJC0JDPVKznfPOzShVfO3ILsvbuXxzVo-1sButvQ"
-	]
```
Comment (cavalle) [inline test/data/invoice-hotel.json:145]: I removed the signatures from the test envelopes as they weren't necessary, and rebuilding the files is (marginally) easier without them.

**Comment by cavalle** [comment]
@samlown we left this pending. I think it's safe to merge, what do you think?

**Comment by samlown** [comment]
@cavalle Sorry! missed this, yes! let's merge. I'll confirm on slack.

File: items_test.go (hunk 1 above)
Comment (samlown) [reply items_test.go:1]: Top.

**Review by samlown — APPROVED** [review]
Looks great!

**Pull request merged**